x-pack/plugins/fleet/server/services/epm/packages/install.ts

```diff
@@ -632,22 +628,60 @@ export const saveKibanaAssetsRefs = async (
   kibanaAssets: Record<KibanaAssetType, ArchiveAsset[]>
 ) => {
   const assetRefs = Object.values(kibanaAssets).flat().map(toAssetReference);
-  await savedObjectsClient.update(PACKAGES_SAVED_OBJECT_TYPE, pkgName, {
-    installed_kibana: assetRefs,
-  });
+  // Because Kibana assets are installed in parallel with ES assets with refresh: false, we almost always run into an
+  // issue that causes a conflict error due to this issue: https://github.com/elastic/kibana/issues/126240. This is safe
+  // to retry constantly until it succeeds to optimize this critical user journey path as much as possible.
+  pRetry(
```

Comment:
Note this retry logic will retry failures other than conflict errors. I think that's probably ok, but happy to fix if we think it's a bad idea.

Reply:
is there a chance 20 retry takes very long in case of trying to retry a persistent error?